app/controllers/instrument_issues_controller.rb

```diff
@@ -8,15 +8,27 @@ class InstrumentIssuesController < ApplicationController
   before_action :init_order_detail
   before_action :init_product
   before_action :init_instrument_issue, only: %i[new create]
+  layout -> { modal? ? false : "application" }
 
   def new
+    redirect_to_order_id = params[:redirect_to_order_id]
+
+    if modal?
+      render partial: "instruments/actions/issues", locals: { product: @product, order_detail: @order_detail, instrument_issue: @instrument_issue, redirect_to_order_id: }
+    end
   end
 
   def create
     @instrument_issue.assign_attributes(create_params)
+    redirect_to_order_id = params[:redirect_to_order_id]
 
     if @instrument_issue.send_notification
-      redirect_to reservations_path, notice: text("create.success")
+      if redirect_to_order_id.present?
```

Comment:
let's move this into a method that returns the redirect path
in that method we can check modal? and ensure we have an order to redirect to 
we can also add a code comment there if needed to clarify the expected code paths